src/app/services/news.service.ts

```diff
@@ -1,57 +1,55 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { Observable, from } from 'rxjs';
+import { Subject, Subscription } from 'rxjs';
 import { map } from 'rxjs/operators';
 import { News } from '../models/News';
+import { RegionSelectService } from './region-select.service';
 
 export interface NewsService {
-  getNews(): Observable<News[]>;
+  getNewsObserver(): Subject<News[]>;
 }
 
 @Injectable({
   providedIn: 'root'
 })
-export class DummyNewsService implements NewsService {
-  getNews(): Observable<News[]>  {
-    return from([[
-      {
-        url: 'http://www.google.com',
-        title: 'COVID kills',
-        description: 'Dumb ppl still not wearing masks.'
-      },
-      {
-        url: 'http://www.bridgebase.com/v3',
-        title: 'Moron dies after injecting himself with bleach',
-        description: 'Trump claims injecting bleach as a COVID treatment is "very, very bad"'
+export class SmartableAiNewsService implements NewsService {
+  private url = 'https://api.smartable.ai/coronavirus/news/';
+  private subscriptionKey = '955eacf9525e45dd8d46a07b7daa649e';
+  private newsObserver: Subject<News[]>;
+  private httpGetSubscription: Subscription;
+
+  constructor(private httpClient: HttpClient, private regionService: RegionSelectService) {
+    this.newsObserver = new Subject<News[]>();
+    this.httpGetSubscription = this.getSubscription(regionService.getRegion());
+    regionService.getRegionObserver().subscribe(
+      region => {
+        this.httpGetSubscription?.unsubscribe();
+        this.httpGetSubscription = this.getSubscription(region);
       }
-    ]]);
+    );
   }
-}
 
-@Injectable({
-  providedIn: 'root'
-})
-export class ActualNewsService implements NewsService {
-  url = 'https://api.smartable.ai/coronavirus/news/';
-  subscriptionKey = '955eacf9525e45dd8d46a07b7daa649e';
-  region = 'global';
-
-  constructor(private httpClient: HttpClient) {}
+  getNewsObserver(): Subject<News[]> {
+    return this.newsObserver;
+  }
 
-  getNews(): Observable<News[]> {
+  private getSubscription(region: string): Subscription {
```

Comment:
Why do you return a subscription and not just the normal observable? surely thats why you have the overly complicated unsubscribe and resubscribe logic above?

Would it be easier to just wrap the `httpClient.get(...)`, and subscribe wherever you call that method?

Reply:
If we change the region, we replace `httpClient.get(...)` with a new one. Currently this is handled by unsubscribing from the old `httpClient.get` and subscribing to a new one.

Reply:
The get method is a finite observable, so it returns one response and then tidies itself up. The source-code completes the observable so there aren't any memory leaks, so you should just be able to call it again without unsubscribing

Check the top answer from this [stack overflow](https://stackoverflow.com/questions/35042929/is-it-necessary-to-unsubscribe-from-observables-created-by-http-methods)

Reply:
OK, no need to unsubscribe then.
We have to change the url between the calls though, so I think we still need to create a new observable for that?

Reply:
don't think you need to worry about creating an observable as such, just consume the one that `HttpClient `returns?